cli/src/commands/explore/calls.rs

```diff
@@ -0,0 +1,145 @@
+        let mut type_examples = print_type_examples(
+            &call.fields,
+            &metadata.runtime_metadata().types,
+            "SCALE_VALUE",
+        )?;
+        type_examples = with_indent(type_examples, 4);
+        let mut output = String::new();
+        write!(output, "Usage:\n    subxt explore {pallet_name} calls {call_name} <SCALE_VALUE>\n        construct the call by providing a valid argument\n\n")?;
+        write!(
+            output,
+            "The call expect expects a <SCALE_VALUE> with this shape:\n{type_description}\n\n{}\n\nYou may need to surround the value in single quotes when providing it as an argument."
+            , &type_examples[4..])?;
+        println!("{output}");
+        return Ok(());
+    }
+
+    // parse scale_value from trailing arguments and try to create an unsigned extrinsic with it:
+    let value = scale_value::stringify::from_str(&trailing_args).0.map_err(|err| eyre!("scale_value::stringify::from_str led to a ParseError.\n\ntried parsing: \"{}\"\n\n{}", trailing_args, err))?;
+    let value_as_composite = value_into_composite(value);
+    let offline_client = new_offline_client(metadata.clone());
+    let payload = tx::dynamic(pallet_name, call_name, value_as_composite);
+    let unsigned_extrinsic = offline_client.tx().create_unsigned(&payload)?;
+    let hex_bytes = format!("0x{}", hex::encode(unsigned_extrinsic.encoded()));
+    println!("Encoded call data:\n    {hex_bytes}");
+
+    Ok(())
+}
+
+fn print_available_calls(pallet_calls: &TypeDefVariant<PortableForm>, pallet_name: &str) -> String {
+    if pallet_calls.variants.is_empty() {
+        return format!("No <CALL>'s available in the \"{pallet_name}\" pallet.");
+    }
+    let mut output = format!("Available <CALL>'s in the \"{pallet_name}\" pallet:");
+
+    let mut strings: Vec<_> = pallet_calls.variants.iter().map(|c| &c.name).collect();
+    strings.sort();
+    for variant in strings {
+        write!(output, "\n    {}", variant).unwrap();
+    }
+    output
+}
+
+fn get_calls_enum_type<'a>(
+    pallet: &'a frame_metadata::v15::PalletMetadata<PortableForm>,
+    registry: &'a PortableRegistry,
+) -> color_eyre::Result<(&'a TypeDefVariant<PortableForm>, &'a Type<PortableForm>)> {
+    let calls = pallet
+        .calls
+        .as_ref()
+        .ok_or(eyre!("The \"{}\" pallet has no calls.", pallet.name))?;
+    let calls_enum_type = registry
+        .resolve(calls.ty.id)
+        .ok_or(eyre!("calls type with id {} not found.", calls.ty.id))?;
+    // should always be a variant type, where each variant corresponds to one call.
+    let calls_enum_type_def = match &calls_enum_type.type_def {
+        TypeDef::Variant(variant) => variant,
+        _ => {
+            return Err(eyre!("calls type is not a variant"));
+        }
+    };
+    Ok((calls_enum_type_def, calls_enum_type))
+}
+
+fn new_offline_client(metadata: Metadata) -> OfflineClient<SubstrateConfig> {
```

Comment:
I don't follow why this hardcoded values are needed

Reply:
It's a bit hard to follow why creating an `offline_client` with hard coded values here which is not obvious to me at least :)

I would prefer if you could rename this function to `mocked_offline_client` and add a comment why/where it's used.